src/_pytest/warnings.py

```diff
@@ -33,10 +35,7 @@ def catch_warnings_for_item(
     """
     config_filters = config.getini("filterwarnings")
     cmdline_filters = config.known_args_namespace.pythonwarnings or []
-    with warnings.catch_warnings(record=True) as log:
-        # mypy can't infer that record=True means log is not None; help it.
-        assert log is not None
-
```

Comment:
Looking at typeshed I think the comment was previously incorrect and the assert wasn't needed https://github.com/python/typeshed/blob/54e1c6ad58961c1faebc6842698eec78f1298f92/stdlib/warnings.pyi#L82
But ye with current logic mypy can't infer `if record -> log is not None` later in the func